src/components/ui/Toggle/Toggle.tsx

```diff
@@ -0,0 +1,57 @@
+import { Input as UIInput, Label as UILabel } from '@faststore/ui'
+import { forwardRef } from 'react'
+import type { InputHTMLAttributes } from 'react'
+import Icon from 'src/components/ui/Icon'
+
+export interface ToggleProps
+  extends InputHTMLAttributes<Omit<HTMLInputElement, 'disabled' | 'type'>> {
+  /**
+   * ID to identify input and corresponding label.
+   */
+  id: string
+  /**
+   * The text displayed to identify the input.
+   */
+  label?: string
+  /**
+   * Specifies that this input should be disabled.
+   */
+  disabled?: boolean
+  variant?: 'horizontal' | 'vertical'
+}
+
+const Toggle = forwardRef<HTMLInputElement, ToggleProps>(function Toggle(
+  { id, label, disabled, variant = 'horizontal', ...otherProps }: ToggleProps,
+  ref
+) {
+  return (
+    <div data-fs-toggle data-fs-toggle-variant={variant}>
+      <div data-fs-toggle-container>
+        <UIInput
+          id={id}
+          ref={ref}
+          type="checkbox"
+          disabled={disabled}
+          aria-label={label ? undefined : `toggle-${id}`}
+          {...otherProps}
+        />
+        <span data-fs-toggle-knob>
```

Comment:
Can we have the size of the text not affect the minimum size of the knob?
![CleanShot 2022-04-27 at 17 12 38](https://user-images.githubusercontent.com/381395/165623193-25a97fdf-4477-4fd2-ba50-d54fd5261255.gif)